ibis/backends/duckdb/__init__.py

```diff
@@ -261,30 +172,176 @@ def register(
+            self._load_extensions(["httpfs"])
             # We don't create a view since DuckDB special cases Arrow Datasets
             # so if we also create a view we end up with both a "lazy table"
             # and a view with the same name
-            with self._safe_raw_sql("SELECT 1") as cursor:
+            with self.con.begin() as con:
                 # DuckDB normally auto-detects Arrow Datasets that are defined
                 # in local variables but the `dataset` variable won't be local
                 # by the time we execute against this so we register it
                 # explicitly.
-                cursor.cursor.c.register(table_name, dataset)
-        elif isinstance(source, (str, Path)):
-            sql, table_name, extensions_required = _generate_view_code(
-                str(source), table_name=table_name, **kwargs
-            )
-            self._load_extensions(extensions_required)
-            self.con.execute(sql)
+                con.connection.c.register(table_name, dataset)
         else:
-            if table_name is None:
-                table_name = next(_gen_table_names)
-            self.con.execute("register", (table_name, source))
+            if any(
+                source.startswith(("http://", "https://")) for source in source_list
+            ):
+                self._load_extensions(["httpfs"])
+            dataset = str(source_list)
+            table_name = table_name or f"ibis_read_parquet_{next(paN)}"
+
+            quoted_table_name = _quote(table_name)
+            sql = f"""CREATE OR REPLACE VIEW {quoted_table_name} AS
+            SELECT * FROM read_parquet({dataset})"""
+
+            self.con.execute(sql)
+
+        return self._read(table_name)
+
+    def read_pandas_or_arrow(self, dataframe, table_name=None):
+        """Register a Pandas DataFrame or pyarrow Table a table in the current database.
+
+        Parameters
+        ----------
+        dataframe
+            The data source.
+        table_name
+            An optional name to use for the created table. This defaults to
+            a sequentially generated name.
+
+        Returns
+        -------
+        ir.Table
+            The just-registered table
+        """
+        table_name = table_name or f"ibis_read_in_memory_{next(pdN)}"
+        self.con.execute("register", (table_name, dataframe))
+
+        return self._read(table_name)
+
+    def read_postgres(self, uri, table_name=None, **kwargs):
+        if table_name is None:
+            raise ValueError(
+                "`table_name` is required when registering a postgres table"
+            )
+        self._load_extensions(["postgres_scanner"])
+        quoted_table_name = _quote(table_name)
+        sql = (
+            f"CREATE OR REPLACE VIEW {quoted_table_name} AS "
+            f"SELECT * FROM postgres_scan_pushdown('{uri}', 'public', '{table_name}')"
+        )
+        self.con.execute(sql)
+
+        return self._read(table_name)
```

Comment:
I guess we're not testing any of this code. Seems like we could spin the `postgres` service up in CI for the duckdb backend to hit this code. Thoughts?

Reply:
Yeah, the old test was just that the appropriate `CREATE VIEW` code was called -- I can add another(?) postgres service for this

Reply:
I don't think you need another service, the existing one is fine, you can add a `services:` array to the `ibis-backends.yml` workflow. There should be some examples in the area.

Reply:
Added a test for this, but after a few rounds of CI shenanigans, it's marked with a `skipif(True)`